File: 0003/P0003.cpp
#include <algorithm>
#include <cmath>
#include <iostream>
#include <vector>

long largestPrimeFactor(long n) {
  std::vector<long> prime_factors;

  for (long i = 2; i < sqrt(n) + 1; ++i) {
    while (n % i == 0) {
      n = n / i;
      prime_factors.push_back(i);
    }
  }

  if (n > 2) {
    prime_factors.push_back(n);
  }

  return *std::max_element(prime_factors.begin(), prime_factors.end());
}

int main() {
  int t{0};

  std::cin >> t;
  for (int i = 0; i < t; ++i) {
    long n{0};
    std::cin >> n;

    long gpf = largestPrimeFactor(n);

    std::cout << gpf << std::endl;
  }

}

File: 0003/P0003.py
import math

def largest_prime_factor(n):
  prime_factors = list()
  for i in range(2, int(math.sqrt(n)) + 1):
    while n % i == 0:
      n = n / i
      prime_factors.append(i)
  if n > 2:
    prime_factors.append(int(n))
  return max(prime_factors)

if __name__ == "__main__":
  t = int(input())
  for i in range(t):
    n = int(input())
    gpf = largest_prime_factor(n)
    print(gpf)